src/test/testHooks.ts

```diff
@@ -1,20 +1,50 @@
 import { Context } from 'mocha';
-import { Telemetry } from '../platform/common/constants';
-import { sendTelemetryEvent } from '../telemetry';
+import { AppinsightsKey, JVSC_EXTENSION_ID, Telemetry } from '../platform/common/constants';
+import TelemetryReporter from '@vscode/extension-telemetry/lib/telemetryReporter';
 import { IS_CI_SERVER } from './ciConstants.node';
+import { extensions } from 'vscode';
+import { sleep } from '../platform/common/utils/async';
+
+let telemetryReporter: TelemetryReporter;
 
 export const rootHooks = {
+    beforeAll() {
+        if (!IS_CI_SERVER) {
+            return;
+        }
+
+        const extensionId = JVSC_EXTENSION_ID;
+        const extension = extensions.getExtension(extensionId)!;
+        const extensionVersion = extension.packageJSON.version;
+
+        // eslint-disable-next-line @typescript-eslint/no-require-imports
+        const reporter = require('@vscode/extension-telemetry').default as typeof TelemetryReporter;
+        telemetryReporter = new reporter(extensionId, extensionVersion, AppinsightsKey, true);
+    },
     afterEach(this: Context) {
         if (!IS_CI_SERVER) {
             return;
         }
 
         let result = this.currentTest?.isFailed() ? 'failed' : this.currentTest?.isPassed() ? 'passed' : 'skipped';
+        const measures = this.currentTest?.duration ? { duration: this.currentTest.duration } : undefined;
         if (this.currentTest?.title) {
-            sendTelemetryEvent(Telemetry.RunTest, this.currentTest?.duration, {
-                testName: this.currentTest?.title,
-                testResult: result
-            });
+            telemetryReporter.sendDangerousTelemetryEvent(
+                Telemetry.RunTest,
+                {
+                    testName: this.currentTest?.title,
+                    testResult: result
+                },
+                measures
+            );
         }
+    },
+    AfterAll: async () => {
```

Comment:
feels weird to have everythign camel cased, and this is pascalCased. I'd doubel check if its `afterAll` instead.

Reply:
Confirmed, its meant to be `afterAll`.